fe/fe-core/src/main/java/org/apache/doris/planner/external/jdbc/JdbcFilterReplace.java

```diff
@@ -0,0 +1,40 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.planner.external.jdbc;
+
+import org.apache.doris.thrift.TOdbcTableType;
+
+public class JdbcFilterReplace {
+    public static String replaceUnsupportedFunctions(TOdbcTableType tableType, String filter) {
+        if (tableType.equals(TOdbcTableType.MYSQL)) {
+            return replaceMySQLFunctions(filter);
+        } else {
+            return filter;
+        }
+    }
+
+    private static String replaceMySQLFunctions(String filter) {
+        if (filter.contains("date_trunc")) {
```

Comment:
Better define a `final static TreeSet<String>`(case insensitive set) to contain these function name. So that we only need to modify that Set to add new function names.